driver-core/src/main/com/mongodb/connection/DefaultClusterFactory.java

```diff
@@ -186,6 +187,20 @@ public Cluster createCluster(final ClusterSettings clusterSettings, final Server
         }
     }
 
+    /**
+     * Creates an embedded Cluster.
+     *
+     * @param embeddedClientSettings the settings
+     * @param embeddedDatabaseFactory the embedded database factory
+     * @return the cluster
+     *
+     * @since 3.8
+     * @mongodb.server.release 4.0
+     */
+    public Cluster createCluster(final MongoClientSettings embeddedClientSettings, final EmbeddedDatabaseFactory embeddedDatabaseFactory) {
+        return new EmbeddedCluster(embeddedClientSettings, embeddedDatabaseFactory);
```

Comment:
Driver core has so much that is protected / namespaced so it may be more than you expect that would get moved.  The idea of running core tests is probably a non starter as so many wont be supported by the embedded API. (admin tasks etc)